src/Npgsql/Schema/DbColumnSchemaGenerator.cs

```diff
@@ -150,7 +150,13 @@ internal ReadOnlyCollection<NpgsqlDbColumn> GetColumnSchema()
                     populatedColumns++;
                 }
 
-                result[i]!.ColumnName = result[i]!.BaseColumnName = field.Name.StartsWith("?column?") ? null : field.Name;
+                Debug.Assert(result != null);
```

Comment:
Suggested the wrong assertion, should be `Debug.Assert(result[i] != null)`, so no warnings should be below. Sorry.

Reply:
@YohDeadfall , I've just applied the assertion `Debug.Assert(result[i] != null); ` and the warnings are still appearing. 

Is the assertion still incorrect?

Reply:
The analyzer isn't so clever as I expected, but probably it's okay for C# since it isn't a memory safe landuage like Rust (: The following compiles as a charm.

```csharp
var column = result[i];
var field = fields[i];

if (column == null)
{
    column = SetUpNonColumnField(field);
    column.ColumnOrdinal = i;
    result[i] = column;
    populatedColumns++;
}

var fieldName = field.Name.StartsWith("?column?") ? null : field.Name;

column.BaseColumnName ??= fieldName;
column.ColumnName = fieldName;
```

Reply:
> The analyzer isn't so clever as I expected

Technically it's the compiler, not an analyzer :trollface: But yeah, IIRC the compiler doesn't track nullability status for array elements, since arrays are arbitrarily large...

Reply:
From Immo's point of view, it's a linter. Compilation is just a side effect (: